libflux/flux-core/src/semantic/infer.rs

```diff
@@ -175,12 +175,72 @@ pub fn equal(
     sub: &mut Substitution,
 ) -> Result<MonoType, Located<Errors<types::Error>>> {
     log::debug!("Constraint::Equal {:?}: {} <===> {}", loc.source, exp, act);
-    exp.try_unify(act, sub).map_err(|error| Located {
-        location: loc.clone(),
-        error,
+    exp.try_unify(act, sub).map_err(|error| {
+        log::debug!("Unify error: {} <=> {}", exp, act);
+
+        Located {
+            location: loc.clone(),
+            error,
+        }
     })
 }
 
+pub(crate) fn temporary_generalize(
```

Comment:
Can you add a doc comment here? How is temporary_generalize different from the normal generalize function?